src/libcore/ops/drop.rs

```diff
@@ -1,85 +1,136 @@
-/// Used to run some code when a value goes out of scope.
-/// This is sometimes called a 'destructor'.
+/// Custom code within the destructor.
 ///
-/// When a value goes out of scope, it will have its `drop` method called if
-/// its type implements `Drop`. Then, any fields the value contains will also
-/// be dropped recursively.
+/// When a value is no longer needed, Rust will run a "destructor" on that value.
+/// The most common way that a value is no longer needed is when it goes out of
+/// scope. Destructors may still run in other circumstances, but we're going to
+/// focus on scope for the examples here. To learn about some of those other cases,
+/// please see [the reference] section on destructors.
 ///
-/// Because of this recursive dropping, you do not need to implement this trait
-/// unless your type needs its own destructor logic.
+/// [the reference]: https://doc.rust-lang.org/reference/destructors.html
 ///
-/// Refer to [the chapter on `Drop` in *The Rust Programming Language*][book]
-/// for some more elaboration.
+/// The `Drop` trait provides a way to customize the destructor of a type.
```

Comment:
This is the first place where `Drop` comes up -- and I feel the best way to explain what it does is to actually have that text you have below, right here instead:
```
/// Destructors consist of two components:
/// - If the special `Drop` trait is implemented for the type of the value, a call to `Drop::drop`.
/// - The automatically generated "drop glue", that recursively calls the destructors
///     of the all fields of this value.
///
/// In other words, the `Drop` trait lets us add our own code to the destructor of a type. [...]
```
Right now, the text just very vaguely says `Drop` is "a way to customize the destructor", but is it the only way and how is it customized anyway? It's not complicated, so I feel we can just say it upfront instead of leaving the reader wonder.

Reply:
:+1: I moved that section upwards  and slightly rewrote the "Using the `Drop` trait" section.